R/autopeek.R

```diff
@@ -0,0 +1,93 @@
+#' Get descriptive one-line summary of an object
+#'
+#' This function is created as a modification of [utils::str()] function. It is supposed to
+#' create more compacted yet informative summary about an object. It's default value of
+#' "icecream.peeking.function"
+#'
+#' @param object The object to be summarized.
+#' @param ... Other arguments passed to methods.
+#'
+#' @details This is a generic function. Default method simply calls `utils::str` function.
+#'
+#' @return The function is mainly used for its side effects -- outputting to the terminal.
+#' However, it also returns an invisible string of the printed summary.
+#'
+#' @seealso [utils::str()] [ic_peek()]
+ic_autopeek <- function(object, ...) UseMethod("ic_autopeek")
+
+#' @export
+ic_autopeek.default <- str
```

Comment:
Roxygen has exported this as a regular function rather than recognising it as an S3 method (see `NAMESPACE`). It looks like this is because Roxygen doesn't parse it correctly. We can remedy this by writing it so it looks more like a regular function:

``` r
#' @export
ic_autopeek.default <- function(object, ...) {
  str(object)
  # Should this be `str(object, ...)`?
}
```

or by tagging it by hand:

``` r
#' @method ic_autopeek default
#' @export
ic_autopeek.default <- str
```

Reply:
Thanks for pointing this out, fixing the issue in f30dc09